File: DemoMMOUnity/Assets/Game/Net/Script/Net/CsProtocol/Protocol/PhysicalPower/PhysicalPowerRequest.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using ZJYFrameWork.Collection.Reference;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Net.CsProtocol.Buffer;

namespace ZJYFrameWork.Net.CsProtocol.Protocol.PhysicalPower
{
    /// <summary>
    /// 获取体力
    /// 
    /// </summary>
    public class PhysicalPowerRequest : Model, IPacket, IReference
    {
        public long uid { get; set; }

        public short ProtocolId()
        {
            return 1023;
        }

        public void Clear()
        {
            uid = 0;
        }

        public static PhysicalPowerRequest ValueOf(long uid)
        {
            var packet = ReferenceCache.Acquire<PhysicalPowerRequest>();
            packet.Clear();
            packet.uid = uid;
            return packet;
        }
    }

    public class PhysicalPowerRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1023;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                return;
            }

            var data = (PhysicalPowerRequest)packet;
            var packetData = new ServerMessageWrite(protocolId: ProtocolId(), data);
            var jsonString = JsonConvert.SerializeObject(packetData);
            buffer.WriteString(jsonString);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            var packet = ReferenceCache.Acquire<PhysicalPowerRequest>();
            return packet;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/physicalPower/physicalPowerRequest.lua
---@class PhysicalPowerRequest
local PhysicalPowerRequest = class("PhysicalPowerRequest")
function PhysicalPowerRequest:ctor()
    --- 玩家uid传过来，可能会用到
    ---@type number
    self.uid = 0
end

---@param uid number 玩家uid传过来，可能会用到
---@return PhysicalPowerRequest
function PhysicalPowerRequest:new(uid)
    self.uid = uid --- long
    return self
end

---@return number
function PhysicalPowerRequest:protocolId()
    return 1023
end

---@return string
function PhysicalPowerRequest:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            uid = self.uid
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return PhysicalPowerRequest
function PhysicalPowerRequest:read(data)

    local packet = self:new(
            data.uid)
    return packet
end

--- 玩家uid传过来，可能会用到
---@return number 玩家uid传过来，可能会用到
function PhysicalPowerRequest:getUid()
    return self.uid
end


return PhysicalPowerRequest
